catalog/ui/catalog-ui-search/src/main/java/net/opengis/filter/v_2_0/PropertyIsLikeType.java

```diff
@@ -0,0 +1,756 @@
+
+  public boolean isSetExpression() {
+    return ((this.expression != null) && (!this.expression.isEmpty()));
+  }
+
+  public void unsetExpression() {
+    this.expression = null;
+  }
+
+  /**
+   * Gets the value of the wildCard property.
+   *
+   * @return possible object is {@link String }
+   */
+  public String getWildCard() {
+    return wildCard;
+  }
+
+  /**
+   * Sets the value of the wildCard property.
+   *
+   * @param value allowed object is {@link String }
+   */
+  public void setWildCard(String value) {
+    this.wildCard = value;
+  }
+
+  public boolean isSetWildCard() {
+    return (this.wildCard != null);
+  }
+
+  /**
+   * Gets the value of the singleChar property.
+   *
+   * @return possible object is {@link String }
+   */
+  public String getSingleChar() {
+    return singleChar;
+  }
+
+  /**
+   * Sets the value of the singleChar property.
+   *
+   * @param value allowed object is {@link String }
+   */
+  public void setSingleChar(String value) {
+    this.singleChar = value;
+  }
+
+  public boolean isSetSingleChar() {
+    return (this.singleChar != null);
+  }
+
+  /**
+   * Gets the value of the matchCase property.
+   *
+   * @return possible object is {@link Boolean}
+   */
+  public Boolean getMatchCase() {
+    return matchCase;
+  }
+
+  /**
+   * Sets the value of the matchCase property.
+   *
+   * @param value allowed object is {@link Boolean}
+   */
+  public void setMatchCase(Boolean value) {
+    matchCase = value;
+  }
+
+  public boolean isSetMatchCase() {
+    return this.matchCase != null;
+  }
+
+  /**
+   * Gets the value of the escapeChar property.
+   *
+   * @return possible object is {@link String }
+   */
+  public String getEscapeChar() {
+    return escapeChar;
+  }
+
+  /**
+   * Sets the value of the escapeChar property.
+   *
+   * @param value allowed object is {@link String }
+   */
+  public void setEscapeChar(String value) {
+    this.escapeChar = value;
+  }
+
+  public boolean isSetEscapeChar() {
+    return (this.escapeChar != null);
+  }
+
+  public String toString() {
+    final ToStringStrategy2 strategy = JAXBToStringStrategy.INSTANCE;
+    final StringBuilder buffer = new StringBuilder();
+    append(null, buffer, strategy);
+    return buffer.toString();
+  }
+
+  public StringBuilder append(
+      ObjectLocator locator, StringBuilder buffer, ToStringStrategy2 strategy) {
+    strategy.appendStart(locator, this, buffer);
+    appendFields(locator, buffer, strategy);
+    strategy.appendEnd(locator, this, buffer);
+    return buffer;
+  }
+
+  public StringBuilder appendFields(
+      ObjectLocator locator, StringBuilder buffer, ToStringStrategy2 strategy) {
+    super.appendFields(locator, buffer, strategy);
+    {
+      List<JAXBElement<?>> theExpression;
+      theExpression = (this.isSetExpression() ? this.getExpression() : null);
+      strategy.appendField(
+          locator, this, "expression", buffer, theExpression, this.isSetExpression());
+    }
+    {
```

Comment:
❓ Why the curly braces in this method?

Reply:
This class came from another library, in which it was auto-generated with an xml binding tool. We needed to fix it to account for a missing xml attribute that is allowed by the schema. There is a TODO at the top of the class to replace class once a proper xml binding library is found.

Reply:
I think in this case the issue is the schema is not wrong. It's correct. The JAXB bindings are wrong. Hence our need for only overriding one of its classes. I think this works as-is. If we need to enhance comments at the top to make it clearer as to what's going on, we should definitely do that.

Reply:
Wait a minute that doesn't make sense. If these were auto-generated classes from the schema, where did the error occur? Looking at the official schema at http://schemas.opengis.net/filter/2.0/filter.xsd if you search for `PropertyIsLikeType` you'll see the attribute is there: 
```
<xsd:complexType name="PropertyIsLikeType">
  <xsd:complexContent>
    <xsd:extension base="fes:ComparisonOpsType">
      <xsd:sequence>
        <xsd:element ref="fes:expression" minOccurs="2" maxOccurs="2"/>
      </xsd:sequence>
      <xsd:attribute name="wildCard" type="xsd:string" use="required"/>
      <xsd:attribute name="singleChar" type="xsd:string" use="required"/>
      <xsd:attribute name="escapeChar" type="xsd:string" use="required"/>
      <xsd:attribute name="matchCase" type="xsd:boolean" use="optional" default="true"/>
    </xsd:extension>
  </xsd:complexContent>
</xsd:complexType>
```

Reply:
They might have used the wrong schemas for code generation.

Reply:
I don't think that is the right answer.  Checking in generated code seems to be a nightmare.  There are precedents of us generating the code during the build.  If the schema is right, we can work even less hard.  Note the ReadMe here: http://schemas.opengis.net/filter/ReadMe.txt

It seems like the schema was fixed last June to add a missing attribute.  I would prefer not to rely on a library partially, and generated code partially.  And I certainly don't like generated code being checked in

Reply:
There is a plan to pull schema-derived classes out to their own repo. Hopefully that will happen sooner rather than later.

Reply:
@coyotesqrl are you saying you approve of this submission because it will be cleaned up later?

Reply:
We don't generally check in generated code (in fact, I don't think we do anywhere) so I don't think it's a great workaround right here to do it as a one-off fix. I was just providing a little extra background information so if it turned out the workaround _was_ the best solution for the short-term, we would at least be comforted by the fact it would be fixed down the road.